### DISCUSSION TITLE
Next/navigation get previous pathname.

### DISCUSSION THREAD

**Discussion by s1139341TimDeRuijter** [vercel/next.js#61258] [Ideas]
### Goals

1. This allows you to see if there is a route to navigate back to.
2. This allows you to see if the previous route is the route you want the user to go back to. So you can have a back button and if the previous route is not the route you want the user to go back to then you could call router.push(""). and when it is, you can call router.back().

### Non-Goals

Does not include the prevention of going back. Which would be a good addition nonetheless.

### Background

I've seen this issue: https://github.com/vercel/next.js/discussions/36723 but those answers don't work because they don't assume a user can use the back and next buttons from the browser. And the user can also enter a route in the navigation bar that is not considered.

### Proposal

Something like this: usePreviousPathname(); And then when there is no previous route to go back to or the previous route is not your application just set it be null.

**Comment by coredevel** [comment]
+1 to this

Did you find a solution?

**Comment by Martin-K-Kamir** [comment]
+1 to this

I’m having the same problem, tried some solutions but can’t find one that works.
